go/runtime/host/loadbalance/loadbalance.go

```diff
@@ -0,0 +1,218 @@
+// Package loadbalance implements a runtime provisioner that internally load-balances requests among
+// multiple runtime instances. This is especially useful on client nodes handling queries.
+package loadbalance
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/oasisprotocol/oasis-core/go/common"
+	"github.com/oasisprotocol/oasis-core/go/common/node"
+	"github.com/oasisprotocol/oasis-core/go/common/pubsub"
+	"github.com/oasisprotocol/oasis-core/go/runtime/host"
+	"github.com/oasisprotocol/oasis-core/go/runtime/host/protocol"
+)
+
+// Config is the configuration for the runtime load balancer.
+type Config struct {
+	// NumInstances is the number of runtime instances to provision.
+	NumInstances int
+}
+
+type lbRuntime struct {
+	id        common.Namespace
+	instances []host.Runtime
+
+	lastIdx atomic.Uint64
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) ID() common.Namespace {
+	return lb.id
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) GetInfo(ctx context.Context) (*protocol.RuntimeInfoResponse, error) {
+	return lb.instances[0].GetInfo(ctx)
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) GetCapabilityTEE() (*node.CapabilityTEE, error) {
+	// TODO: This won't work when registration of all client runtimes is required.
+	return lb.instances[0].GetCapabilityTEE()
```

Comment:
The use case is for queries so those CapabilityTEEs for those other instances are ignored anyway.

As noted in the TODO this will be a problem once instances need to publish their RAKs to be able to communicate with the key manager.

Reply:
> The use case is for queries so CapabilityTEEs for those other instances are ignored anyway.

Since they are ignored, we could probably disable attestations for them.

TODO is ok and can be addressed at a latter time. I would first test this in production to see if this works :)

Reply:
Disabling attestations would make it impossible for them to communicate with key managers.